### ISSUE TITLE
vault_approle_auth_backend_role does not work with vault v1.20

### ISSUE THREAD

**Issue by kinghrothgar** [terraform-providers/terraform-provider-vault#485]
Attempting to apply simple vault_approle_auth_backend_role consistantly fails and regularly causes panic.

Similarly importing a simple approle role causes failure and panic

### Terraform Version
Terraform v0.12.6
provider.vault v2.1.0
vault v1.20

### Affected Resource(s)
- vault_approle_auth_backend_role

### Terraform Configuration Files
```resource "vault_approle_auth_backend_role" "staging_grpc_ssl_example" {
  depends_on = [
    "vault_policy.staging_grpc_ssl_example"
    ,"vault_auth_backend.approle_production"
  ]
  backend            = "${vault_auth_backend.approle_staging.path}"
  role_name          = "grpc_ssl_example"
  policies           = ["staging_grpc_ssl_example"]
}
```

### Debug Output
Output of running `terraform apply` https://gist.github.com/kinghrothgar/7e52a1c26a78c4d1a86576d41a9c3814

### Panic Output
crash.log from import https://gist.github.com/kinghrothgar/ee400f0d2212c4914c9e3878d340a103
crash.log from apply https://gist.github.com/kinghrothgar/a5a246d5c4e3145e50efcd541c209ce8

### Steps to Reproduce
1. Create a tf with vault_approle_auth_backend_role
2. `terraform apply`

**Comment by kinghrothgar** [comment]
I just tested my terraform files on `v1.1.5` and it works. There are approle changes mentioned in `v1.2.0` changelog so I assume that's what broke this / the GRPC spec

**Comment by lawliet89** [comment]
Should have been fixed by https://github.com/terraform-providers/terraform-provider-vault/pull/47

You might want to try ti again with a build from master branch or wait for the next release.

**Comment by kinghrothgar** [comment]
This is fixed in `v2.0.0`
